homeassistant/components/conversation/default_agent.py

```diff
@@ -332,7 +338,15 @@ async def _build_speech(
         for state in all_states:
             device_class = state.attributes.get("device_class", "_")
             key = f"component.{state.domain}.entity_component.{device_class}.state.{state.state}"
-            state.state = translations.get(key, state.state)
+            attributes = ReadOnlyDict(
+                state.attributes
+                | {
+                    ATTR_TRANSLATED_STATE: translations.get(key, state.state),
+                    ATTR_UNTRANSLATED_STATE: state.state,
+                }
+            )
+            state.attributes = attributes
+            state.state = str(state.attributes.get(ATTR_TRANSLATED_STATE))
```

Comment:
```suggestion
            state.state = str(state.attributes[ATTR_TRANSLATED_STATE])
```

Any reason you can think not to do this?

Reply:
Not really, no. I'll change it.